### PR TITLE
Dont return string paths

### PR DESCRIPTION
**Pull request by faissaloo**
(the commit history got a bit weird)

### CODE REVIEW

File: LibGUI/GFilePicker.h [hunk 1]
```diff
@@ -9,12 +10,12 @@ class GFilePicker final : public GDialog {
     virtual ~GFilePicker() override;
 
     // TODO: Should this return a FileSystemPath instead?
```
Comment (rburchell) [inline LibGUI/GFilePicker.h]: TODO should be removed.

**Comment by faissaloo** [comment]
I'm cautious of rebasing because I don't know if it's going to do something funny on merge.

**Comment by awesomekling** [comment]
Hi faissaloo, this PR contains old commits that belong to your previous PR, so we need to clean this up.
Please update the branch to only contain the relevant new commit :)

Learning how to use `git rebase -i` and squashing commits together is really important, so please look into it. Here's a quick primer for example: http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html

**Comment by awesomekling** [comment]
Oh also, please provide some more context in commit messages to keep the git log understandable.
For example this commit could be something like "GFilePicker: Return paths as FileSystemPath rather than String."

**Comment by faissaloo** [comment]
I did say that would happen...

**Comment by awesomekling** [comment]
When you pull and have local commits, you can use `git pull --rebase` to avoid automatic merging behavior.

**Comment by faissaloo** [comment]
I just reset it to master and then cherry picked it

**Comment by awesomekling** [comment]
Great! And this is a nice change, thanks for doing it :)

**Pull request merged**